timestamp-implementation-advice.xml

```diff
@@ -85,87 +85,129 @@
+        this document, we therefore define three different kinds of
+        clocks that a system may or may not provide.</t>
+      </t>
+      <section title="Raw Clock">
+        <t>
+          At its most fundamental, a system may provide access to its own,
+          unadultered perception of time, its "raw clock." This clock is
+          typically measuring time by counting cycles of an oscillator. Its
+          quality therefore relies on the stability of this oscillator.
+        </t><t>
+          As it is a purely subjective time; no general meaning can be
+          attached to any specific value. Only the amount of time passed
+          can be determined by comparing two values.
+        </t><t>
+          Because the raw clock is an unaltered counter, its is strictly
+          monotonically increasing. Its values will always grow at a steady
+          pace, never decrease, never make unexpected jumps, or stop.
+
+          <!-- Would it be fair to say that the raw clock is linear? -->
+        </t>
+      </section>
+      <section title="Adjusted Raw Clock">
+        <t>
+          Even if highly accurate oscillators are used, a raw clock measures
+          time at a rate slightly differening from  wall time. This difference
+          is called "clock drift." It depends not only on the quality of the
+          time source but also on environmental factors such as temperature.
+          That is, drift will not be constant for a given clock but may
+          increase or decrease over time.
+        </t><t>
+          When this drift is componsated by comparing the time of the raw clock
+          to some external time source that is considered to be closer to wall
+          time, the result is the "adjusted raw clock." This adjustment doesn't
+          happen sporadically but rather, the rate of advance of time is slowed
+          down or sped up slightly until it approaches the reference time again.
+          As a result, adjusted raw time is still strictly monotonic
+          increasing. Like the raw clock, the adjusted raw clock is subjective
+          with no specific meaning attached to its values.
+        </t><t>
+          Since it can only work with an external time source, the adjusted
+          raw clock may is susceptible to vulnerabilites of these sources,
+          which may be exploited to maliciously manipulate this time.
+        </t>
+      </section>
+      <section title="Real Time Clock">
+        <!-- Maybe we need a better name now? -->
+        <t>
+          With the adjusted raw clock, a system already has access to a time
+          source that passes at a rate very similar to wall time. By
+          adjusting the time value so that it represents the time passed
+          since an epoch, a well-defined point of wall time such as seconds
+          since midnight January 1st, 1970 on Unix systems, time values
+          themselves gather meaning. The result is a "real time clock."
```

Comment:
Use of comma's to give a definition of epoch is perhaps a bit unclear?
Maybe we should use brackets or dashes instead?